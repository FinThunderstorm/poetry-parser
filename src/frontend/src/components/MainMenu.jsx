import React from 'react'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import Typography from '@mui/material/Typography'
import { Button } from '@mui/material'
import { Link } from 'react-router-dom'

const MainMenu = () => {
    const appName = 'poetry-parser'
    return (
        <>
            <AppBar color="primary">
                <Toolbar>
                    <Typography
                        variant="h6"
                        component="div"
                        sx={{ flexGrow: 1 }}
                    >
                        {appName}
                    </Typography>
                    <Button component={Link} to="/" color="inherit">
                        Home
                    </Button>
                    <Button component={Link} to="/upload" color="inherit">
                        Upload
                    </Button>
                </Toolbar>
            </AppBar>
            <Toolbar />
        </>
    )
}

export default MainMenu
