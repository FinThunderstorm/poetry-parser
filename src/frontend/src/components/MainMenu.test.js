/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'

import { MemoryRouter, Route, Routes } from 'react-router-dom'
import MainMenu from './MainMenu'

describe('MainMenu', () => {
    const MainMenuWrapper = ({ url }) => (
        <MemoryRouter initialEntries={[url]}>
            <Routes>
                <Route path="/" element={<MainMenu />} />
            </Routes>
        </MemoryRouter>
    )

    test('MainMenu renderes correctly', () => {
        render(<MainMenuWrapper url="/" />)

        const title = screen.getByText('poetry-parser', {
            exact: false,
        })
        const home = screen.getByText('home', {
            exact: false,
        })
        const upload = screen.getByText('upload', {
            exact: false,
        })
        expect(title).toBeDefined()
        expect(home).toBeDefined()
        expect(upload).toBeDefined()

        const tree = renderer.create(<MainMenuWrapper url="/" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
