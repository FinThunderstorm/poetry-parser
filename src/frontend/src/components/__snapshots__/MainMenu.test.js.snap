// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu MainMenu renderes correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        poetry-parser
      </div>
      <a
        className="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Home
      </a>
      <a
        className="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
        href="/upload"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Upload
      </a>
    </div>
  </header>,
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
  />,
]
`;
